### DISCUSSION TITLE
How to change the default roms folder (to internal storage) if the TF2/EXT slot on RG280V is physically defective?

### DISCUSSION THREAD

**Discussion by aliquantulus** [eduardofilo/RG350_adam_image#337] [Hardware problems?]
### What happened?

I received my first Anbernic device this week, but no matter what I do, the secondary TF2/EXT microSD slot on my RG280V seems to be defective. I have tried 3 different software packages to format the card to FAT32 with no label (guiformat.exe, Raspberry Pi Imager and GParted) and 4 different microSD cards (ranging from 16 GB to 128 GB), but no matter what I do, I can only see the  mmcblk0 (nothing happens in dmesg (via USB 10.1.1.2.) and lsblk only shows  mmcblk0 and it's two partitions (when using Adam image 2.1 and 1.6). I get the exact same results from the slot when I boot the original Anbernic Kioxia SD-card. I have reached out to Anbernic, but if I could get games to be scanned from slot1 I would consider keeping this defective device (at the moment I can manually search for roms via retroarch which is the only "game" found, but that's a lot more steps to start gaming each time than just having the device scan the correct folder and finding the games that way).

### Device

RG280V

### Version

v2.1

**Comment by eduardofilo** [comment]
There are many adjustments made in Adam to make things work towards the external card. I'll tell you the main ones if you want to give it a try, but there will probably be things I've left out that you'll have to figure out.

* For SimpleMenu, edit the .ini files in `~/.simplemenu/section_groups`
* For RetroArch, edit the `.retroarch/retroarch.cfg`

Another thing you could try is to create a new partition on the internal card and then mount it manually on `/media/sdcard` with some script during boot.

In the past I remember one person also had problems with the EXT slot and was able to fix it by reflowing the solder joints in the slot.

**Reply by aliquantulus** [reply]
I actually tried creating the /media/sdcard folder (with the intent of putting the roms there), but even with sudo I got read only file system error when I tried sudo mkdir /media/sdcard/. If you have a way around that error then that will probably be the easiest fix for my RG280V and I would be eternally grateful (I'll even donate to get a fix where I can use a 128 GB microSD with roms in slot #1.

**Reply by Sakitoshi** [reply]
i have mine working with only a single sd card.
i created a script to mount _mmcblk0p3_ (AKA a third partition, fat32 in this case) on _/media/data/local/etc/init.d/_ via terminal (root permission required, couldn't do it via ftp) and named it _S06_loadroms.sh_, name doesn't really matter, but needs to start with _S06_ to load it after _S05_loadexfat.sh_. the content is:
```
#!/bin/sh

case "$1" in
    start)
        printf "Loading roms partition"
        if [ ! -d /media/sdcard ] && mount|grep -q /dev/mmcblk0p3
        then
        mount -o remount,rw /media
        umount /media/mmcblk0p3
        rmdir /media/mmcblk0p3
        mkdir -p /media/sdcard
        mount /dev/mmcblk0p3 -o rw,uid=1000,gid=100 /media/sdcard
        mount -o remount,ro /media
        echo "done"
        fi
        ;;
    stop)
        # Nothing to do
        ;;
    *)
        echo "Usage: $0 {start|stop}"
        exit 1
        ;;
esac

```
the script first checks if a mount point named _sdcard_ already exists (if it does it means a second sdcard is inserted and does nothing) and if it doesn't it remounts the third partition of the first as _sdcard_ and sets the proper permissions.
you can also just create an exfat partition and label it _sdcard_ (all lowercase), the mount point of a labeled partition is named after the label instead of generic names and as the system just looks for a mount point named _sdcard_, the system is happy to use it without complains as long as it's lowercase. exfat supports lowercase labels while fat32 labels are always uppercase.

i find kinda dumb that you are forced to use a second sdcard when a single sdcard can be used for the system and roms without issues.

**Comment by aliquantulus** [comment]
After trying a lot of different folders (and partitioning a separate partition that auto mounted to /media/mmcblk0p3 etc) I finally gave up and opened it up. The pads and solder joints were perfect, but that little thing on the side of TF-slot2 that registers if a card is inserted or not was bent towards slot#1 (while the metal thing on slot#1 was flush with the rest of the card-slot), so I simply bent that parallell and now it has created all the folders on the secondary SD-card as expected, and when I put roms on that card everything works like it should. That said, I really wish it would be easier (or at least not seemingly impossible) to also put roms on the internal card. It could be a fallback when a card slot fails (and right to repair mentaly would suggest that a secondary card slot being faulty should not make the entire device useless).
